pyterrier/apply.py

```diff
@@ -205,10 +214,21 @@ def _fn(iterdict):
     if iter:
         if kwargs.get("add_ranks", False):
             raise ValueError("add_ranks=True not supported with iter=True")
-        return ApplyGenericIterTransformer(fn, *args, batch_size=batch_size, **kwargs)
-    return ApplyGenericTransformer(fn, *args, batch_size=batch_size, **kwargs)
-
-def by_query(fn : Union[Callable[[pd.DataFrame], pd.DataFrame], Callable[[pt.model.IterDict], pt.model.IterDict]], *args, batch_size=None, iter=False, verbose=False, **kwargs) -> pt.Transformer:
+        rtr = ApplyGenericIterTransformer(fn, *args, batch_size=batch_size, **kwargs)
+    else:
+        rtr = ApplyGenericTransformer(fn, *args, batch_size=batch_size, **kwargs)
+    if transform_outputs is not None:
+        _bind(rtr, transform_outputs, as_name='transform_outputs')
```

Comment:
Are you sure we need bind here? Can't we pass transform_outputs straight through as a kwarg?

I guess that means it wouldn't take self as an argument, but I don't think you ever really need a reference back to self for an apply.